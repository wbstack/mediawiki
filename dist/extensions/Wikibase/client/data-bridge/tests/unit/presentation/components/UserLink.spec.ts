import UserLink from '@/presentation/components/UserLink.vue';
import { shallowMount } from '@vue/test-utils';

describe( 'UserLink', () => {
	it( 'links to the user page if given a positive user ID', () => {
		const userId = 1234;
		const userName = 'Example User';
		const router = {
			getPageUrl: jest.fn( ( title ) => `https://wiki.example/wiki/${title}` ),
		};
		const wrapper = shallowMount( UserLink, {
			propsData: { userId, userName, router },
		} );
		expect( wrapper.element ).toMatchSnapshot();
		expect( router.getPageUrl ).toHaveBeenCalledWith( `Special:Redirect/user/${userId}` );
	} );

	it( 'inserts only the user name if given a zero user ID', () => {
		const userName = 'Example User';
		const router = { getPageUrl: jest.fn() };
		const wrapper = shallowMount( UserLink, {
			propsData: { userId: 0, userName, router },
		} );
		expect( wrapper.element ).toMatchSnapshot();
		expect( router.getPageUrl ).not.toHaveBeenCalled();
	} );
} );
