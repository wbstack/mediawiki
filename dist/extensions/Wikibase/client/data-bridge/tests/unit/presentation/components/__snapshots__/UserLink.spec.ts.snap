// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLink inserts only the user name if given a zero user ID 1`] = `
<span>
  <bdi>
    Example User
  </bdi>
</span>
`;

exports[`UserLink links to the user page if given a positive user ID 1`] = `
<span>
  <a
    href="https://wiki.example/wiki/Special:Redirect/user/1234"
  >
    <bdi>
      Example User
    </bdi>
  </a>
</span>
`;
